### PR TITLE
Remove unnecessary copy of interface map

### PR DESCRIPTION
**Pull request by karthikreddy09**
Signed-off-by: The Mathworks Inc <Roy.Lurie@mathworks.com>

### CODE REVIEW

File: framework/src/bundle/BundleContext.cpp [hunk 1]
```diff
@@ -297,17 +297,10 @@ InterfaceMapConstPtr BundleContext::GetService(
   // but we ignore it since the time window is small and
   // the result is the same as if the calling thread had
   // won the race condition.
-
-  // Although according to the API contract the returned map should not be modified, there is nothing stopping the consumer from
-  // using a const_pointer_cast and modifying the map. This copy step is to protect the map stored within the framework.
```
Comment (saschazelzer) [inline framework/src/bundle/BundleContext.cpp]: Why is that comment not valid anymore? Are you not concerned about clients anymore const-casting the map? Would be fine with me.

Comment (karthikreddy09) [reply framework/src/bundle/BundleContext.cpp]: It's similar to the scenario where an API returns a const reference and the user const_cast's it and modifies the object. At that point, it's a breach of contract and the user has to deal with the consequences/undefined behavior.

Comment (saschazelzer) [reply framework/src/bundle/BundleContext.cpp]: Well, I agree. I believe it was like that before we introduced the copy. So the concerns in the comment are gone, right?

Comment (karthikreddy09) [reply framework/src/bundle/BundleContext.cpp]: yes

**Pull request merged**